server/user-groups.ts

```diff
@@ -246,3 +287,42 @@ export class GlobalAuth extends Auth {
 		return true;
 	}
 }
+
+export const Permissions = new class {
+	approvedPermissions(room: Room | null = null) {
+		// support some config permission groups that aren't already a command name - these need to be hardcoded
+		// since there's no good way to automate this
+		const ALLOWED_COMMANDS = [
+			"showmedia", "broadcast", "tournaments", "gamemoderation", "gamemanagement", "minigame", "game",
+		];
+		for (const cmd in Chat.commands) {
+			const entry = Chat.commands[cmd];
+			if (typeof entry !== 'function') continue;
+			// remove room-plugin commands unless it is that room
+			if (entry.roomSpecific && entry.roomSpecific !== room?.roomid) continue;
+			// Assume if it passes a room into CommandContext.can that it's roomonly
+			if (Chat.commands[cmd].toString().includes('room)) return false;')) {
```

Comment:
`room)) return false;` is a pretty awkward check tbh...

Reply:
Perhaps ``` /(this|user).can\((.+), (.+), (.+)\)/.test()```?

Reply:
It would still be weird, but assuming `.` doesn't include newlines, it might be good enough.

I would remove the `(this|user)` part, though; and also your first `.` needs to be escaped.